### PR TITLE
TMEDIA-166 - Search content source encoding

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

Encode the query string value being used to call the search API 

## Jira Ticket
- [TMEDIA-166](https://arcpublishing.atlassian.net/browse/TMEDIA-166)

## Test Steps

1. Checkout this branch `git checkout TMEDIA-166-search-content-source-encoding`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/search-content-source-block`
3. Using content debugger - http://localhost/pagebuilder/tools/debugger?_website=the-gazette&type=content-debugger&source=search-api
4. Using the following value - `vidèo` verify results are now returned
5. Verify you see a search result in the theme search results page - http://localhost/search/vid%C3%A8o/?_website=the-gazette

### Before

Search results not found -

TMEDIA-166-after

### After

Search results now found - 
TMEDIA-166-after

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [X] Confirmed all the test steps a reviewer will follow above are working.
- [X] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [X] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [X] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [X] Confirmed this PR has unit test files
  - [X] Ran `npm run test`, made sure all tests are passing
- [X] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/search-content-source-block/sources/search-api.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ export default {
   resolve(contentOptions) {
     const { query, page, 'arc-site': arcSite } = contentOptions;
     if (query) {
-      return `https://search.arcpublishing.com/search?&q=${query}&page=${page || 1}${arcSite ? `&website_id=${arcSite}` : ''}${SEARCH_KEY ? `&key=${SEARCH_KEY}` : ''}`;
+      return `https://search.arcpublishing.com/search?&q=${encodeURIComponent(decodeURIComponent(query))}&page=${page || 1}${arcSite ? `&website_id=${arcSite}` : ''}${SEARCH_KEY ? `&key=${SEARCH_KEY}` : ''}`;
```
Comment (JackHowa) [inline blocks/search-content-source-block/sources/search-api.js:8]: not sure I understand why the query is being decoded first if we're trying to encode it?

Comment (JackHowa) [inline blocks/search-content-source-block/sources/search-api.js:8]:
```suggestion
      return `https://search.arcpublishing.com/search?&q=${encodeURIComponent(query)}&page=${page || 1}${arcSite ? `&website_id=${arcSite}` : ''}${SEARCH_KEY ? `&key=${SEARCH_KEY}` : ''}`;
```

File: blocks/search-content-source-block/sources/search-api.test.js [hunk 2]
```diff
@@ -16,6 +16,14 @@ describe('the search content source block', () => {
       });
       expect(url).toEqual('https://search.arcpublishing.com/search?&q=test&page=1&website_id=the-sun&key=ABCDEF');
     });
+
+    it('should encode the query value', () => {
+      const url = contentSource.resolve({
+        query: 'vidèo',
```
Comment (JackHowa) [inline blocks/search-content-source-block/sources/search-api.test.js:22]: nice test, helpful to understand an example where this is used

**Review by JackHowa — CHANGES_REQUESTED** [review]
I might be missing the reason for decodeURI. But I was getting the same results without decode testing it locally. Likely not a big deal but curious

File: blocks/search-content-source-block/sources/search-api.js (hunk 1 above)
Comment (matthewroach) [reply blocks/search-content-source-block/sources/search-api.js:8]: We have to encode the query to account for the accent characters being sent to the search API.. But in some cases the query is already encoded, so to account for that we decode and encode it. To test this you can remove the `decodeURIComponent` part of this and then load the search page - http://localhost/search/vid%C3%A8o/?_website=the-gazette and you'll see it's double encoding the query value and returning no results

Comment (JackHowa) [reply blocks/search-content-source-block/sources/search-api.js:8]: `But in some cases the query is already encoded` didn't know this bit

**Review by JackHowa — APPROVED** [review]
Approved. Didn't realize there was an edge case where the query is already encoded coming in

**Pull request merged**